R/guides-.R

```diff
@@ -502,43 +563,47 @@ Guides <- ggproto(
 
     Map(
       grobs    = grobs,
-      position = names(grobs),
+      position = positions,
+      inside_position = inside_positions,
+      inside_just = inside_justs,
       self$package_box,
       MoreArgs = list(theme = theme)
     )
   },
 
   # Render the guides into grobs
-  draw = function(self, theme,
-                  default_position = "right",
-                  direction = NULL,
+  draw = function(self, theme, positions = NULL, direction = NULL,
                   params = self$params,
                   guides = self$guides) {
-    positions <- vapply(
+    positions <- positions %||% vapply(
       params,
-      function(p) p$position[1] %||% default_position,
-      character(1)
+      function(p) p$position[1] %||% "right",
+      character(1), USE.NAMES = FALSE
     )
-    positions <- factor(positions, levels = c(.trbl, "inside"))
-
-    directions <- rep(direction %||% "vertical", length(positions))
     if (is.null(direction)) {
-      directions[positions %in% c("top", "bottom")] <- "horizontal"
+      directions <- ifelse(
+        positions %in% c("top", "bottom"), 
+        "horizontal", "vertical"
+      )
+    } else {
+      directions <- rep(direction, length(positions))
```

Comment:
I'm not understanding this change. The `else` branch repeats `NULL` some times, which remains `NULL` because it has 0-length?

Reply:
My apology, should be fixed